pkg/querier/querier.go

```diff
@@ -565,6 +570,9 @@ func (q querier) mergeSeriesSets(sets []storage.SeriesSet) storage.SeriesSet {
 
 		// SeriesSet may have some series backed up by chunks, and some not.
 		for set.Next() {
+			if ctx.Err() != nil {
+				return nil, ctx.Err()
+			}
```

Comment:
Why we dont return `storage.ErrSeriesSet(err)` here instead of in the caller?

Reply:
Yes, we can do that. Changed